superset-frontend/src/explore/components/controls/DndColumnSelectControl/utils/optionSelector.ts

```diff
@@ -67,9 +67,10 @@ export class OptionSelector {
     return !!this.getValues()?.includes(value);
   }
 
-  getValues(): string[] | string | undefined {
+  getValues(): string[] | undefined {
+    // ensure an array type return value
     if (!this.multi) {
-      return this.values.length > 0 ? this.values[0].column_name : undefined;
+      return this.values.length > 0 ? [this.values[0].column_name] : undefined;
```

Comment:
I assume the current return type `string[] | string | undefined` is there to make it fully compatible with the non-dnd control, which returns `string` when not in `multi` mode. Am I right @kgabryje?

Reply:
i think it's not really about compatibility between dnd and non-dnd, but what chart's logic expects - I think charts expect a string for a single value controls and an array for multi value controls. I tested this PR with Bubble Chart and I'm getting an error after changing a value in a non-multi control
CC @zhaoyongjie 
![image](https://user-images.githubusercontent.com/15073128/131488725-b22fc13b-66f6-46e3-a723-b85405f79b5a.png)

Reply:
Right - I think v1 charts won't mind, but `viz.py` definitely has lots of logic that either assumes control values are either arrays or strings.

Reply:
Sorry, I overlooked that part of the logic. fixed